### DISCUSSION TITLE
patched together workaround to the repo-locking nature of borg2

### DISCUSSION THREAD

**Discussion by SpiritInAShell** [borgbackup/borg#8320] [Show and tell]
tl;dr: not a request, not a howto, most likely not useful, just rambling about some experiments

BTW, I use proxy repositories to cirumvent the repo-locking, especially on a low-bandwidth remote storage. It looks similar to this:

- btrfs subvolumes
- `rcreate --repo=local1.borg2 --other-repo=sftp-remote.borg2` for every subvolume
- **simultanous pushing data to subvolume repo{1,2,3,4,...} from different disks or devices** (directly or transfer, or export/import-tar)
  - doing snapshots of repoX to continue adding to repoX+1 while `transfer` (so some deduplication going on the local space)
- marking local repo as ready to sync
- (manually or) other task cycling through all "ready to sync" repos and doing `transfer` to remote storage.

Now I can prepare the data being sent to remote storage "pre-seeded", ready chunked and compressed.

Very fiddly and prone to errors. Needs much local space. Needs much time. Unneccessary high load on the local drive holding proxy repos. Limited local deduplication. Don't use this ~~in production~~!

in borg2, as there is lossless `export-tar` and `transfer` in borg2, it is now possible to downstream from a central repo selected archives. With `--bypass-lock` on read-only snapshots **this should** also make parallel downstreams from remote possible. Locking nature of borg2 somewhat circumvented.

I wish borg1 could be enhanced to export-tar losslessly. Then exporting to the local repo which is associated to the remote repo would be faster.

**Comment by ThomasWaldmann** [comment]
Closing this as outdated: it has seen no activity for a long time and predates the current borg releases.

If this still affects you on a current version - borg 1.4.x (stable) or the borg 2.0.0 beta - please open a new discussion with up-to-date details.
